swift/Arrow/Sources/Arrow/ArrowReaderHelper.swift

```diff
@@ -21,8 +21,9 @@ import Foundation
 private func makeBinaryHolder(_ buffers: [ArrowBuffer],
                               nullCount: UInt) -> Result<ArrowArrayHolder, ArrowError> {
     do {
-        let arrowData = try ArrowData(ArrowType(ArrowType.ArrowBinary), buffers: buffers,
-                                      nullCount: nullCount, stride: MemoryLayout<Int8>.stride)
+        let arrowType = ArrowType(ArrowType.ArrowBinary)
+        let arrowData = try ArrowData(arrowType, buffers: buffers,
+                                      nullCount: nullCount, stride: ArrowType.getStride(arrowType))
```

Comment:
How about computing `stride:` in `ArrowData.init()` because `ArrowData.init()` can use the given `ArrowType`?